dev/buildtool/base_metrics.py

```diff
@@ -235,128 +236,112 @@ def __init__(self, options):
-    return family.get(labels)
-
-  def inc(self, name, labels, amount=1):
-    """Increment the implied counter."""
-    self.counter(name, labels).inc(amount=amount)
-
-  def register_gauge(self, name, description, label_names, value_type=long):
-    """Defines a gauge metric."""
-    return self.__register_family(name, self._do_make_gauge_family,
-                                  description, label_names, value_type)
-  def _do_make_gauge_family(
-      self, name, description, label_names, value_type):
+  def _do_make_family(
+      self, family_type, name, description, label_names):
     """Creates new metric-system specific gauge family.
 
     Args:
+      family_type: MetricFamily.COUNTER, GUAGE, or TIMER
       name: [string] Metric name.
       description: [string] Metric help description.
       label_names: [list of string] The labels used to distinguish instances.
-      value_type: [type] int, long, or float
 
     Returns:
-      GaugeFamily
+      specialized MetricFamily for the given type and registry implementation.
     """
     raise NotImplementedError()
 
-  def gauge(self, name, labels):
-    """Returns the specified gauge."""
-    family = self.__metric_families[name]
-    if family.family_type != family.GAUGE:
-      raise TypeError('{have} is not a Gauge'.format(have=family))
-    return family.get(labels)
+  def queue_update(self, metric):
+    """Add metric to list of metrics to push out."""
+    with self.__update_mutex:
+      self.__updated_metrics.add(metric)
+
+  def inc_counter(self, name, labels, description, **kwargs):
+    """Track number of completed calls to the given function."""
+    family = self.__ensure_family(
+        MetricFamily.COUNTER, name, labels, description)
+    counter = family.get(labels)
+    counter.inc(**kwargs)
+    return counter
+
+  def count_call(self, name, labels, description,
+                 func, *pos_args, **kwargs):
+    """Track number of completed calls to the given function."""
+    labels = dict(labels)
+    success = False
+    try:
+      result = func(*pos_args, **kwargs)
+      success = True
+      return result
+    finally:
+      labels['success'] = success
+      self.inc_counter(name, labels, description, **kwargs)
 
-  def set(self, name, labels, value):
+  def set(self, name, labels, description, value):
     """Sets the implied gauge with the specified value."""
-    self.gauge(name, labels).set(value)
+    family = self.__ensure_family(
+        MetricFamily.GAUGE, name, labels, description)
+    gauge = family.get(labels)
+    gauge.set(value)
+    return gauge
 
-  def track(self, name, labels, func, *pos_args, **kwargs):
+  def track_call(self, name, labels, description, func, *pos_args, **kwargs):
     """Track number of active calls to the given function."""
-    self.gauge(name, labels).track(func, *pos_args, **kwargs)
+    family = self.__ensure_family(
+        MetricFamily.GAUGE, name, labels, description)
+    gauge = family.get(labels)
```

Comment:
These two `family` and `gauge|counter|timer` calls appear several places, maybe a helper function? Maybe one just for `gauge`?